test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
OK, fair enough. I was thinking to use this replace functionality in tests where we can replace active "dity" container with a new "pristine" one. Right now, it looks like a better idea would be to implement "container.clear()" for this usage scenario.